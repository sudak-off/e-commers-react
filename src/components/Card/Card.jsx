import React, {useState} from 'react';

import styles from './Card.module.scss'

const Card = ({id, title, imageUrl, price, onFavorite, onPlus, favorited = false}) => {
    const [isAdded, setIsAdded] = useState(false);
    const [isFavorite, setIsFavorite] = useState(favorited);

    const onClickPlus = () => {
        onPlus({title, imageUrl, price})
        setIsAdded(!isAdded)
    }

    const onClickFavorite = () => {
        onFavorite({id, title, imageUrl, price})
        setIsFavorite(!isFavorite)
    }

    const numbFnt = (num) => {
        return num.toLocaleString('ua-UA');
    }


    return (
        <div className={styles.card}>
            <div className={styles.favorite} onClick={onClickFavorite}>
                <img src={!isFavorite ? "/img/heart-unliked.svg" : "/img/heart-like.svg"} alt="like"/>
            </div>
            <img width={133} height={112} src={imageUrl} alt="shoes"/>
            <h5>{title}</h5>
            <div className='d-flex justify-between align-center'>
                <div className='d-flex flex-column'>
                    <span>Цена:</span>
                    <b>{numbFnt(price)} грн.</b>
                </div>
                <button>
                    <img className={styles.plus} onClick={onClickPlus}
                         src={isAdded ? '/img/btn-checked.svg' : '/img/btn-plus.svg'} alt="plus"/>
                </button>
            </div>
        </div>
    );
};

export default Card;