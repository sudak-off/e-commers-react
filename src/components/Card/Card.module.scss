.card {
    border: 1px solid #F3F3F3;
    border-radius: 30px;
    padding: 20px;
    width: 210px;
    margin-right: 30px;
    margin-bottom: 30px;
    transition: box-shadow 0.2s ease-in-out, transform 0.2s ease-in-out;

    .plus{
        cursor: pointer;
    }

    &:hover {
        box-shadow: 0 20px 35px rgba(0, 0, 0, 0.06);
        transform: translateY(-5px);
    }

    .favorite {
        position: absolute;
        cursor: pointer;
    }

    button {
        width: 32px;
        height: 32px;
        background: #FFFFFF;
        border: 1px solid #F2F2F2;
        border-radius: 8px;
        cursor: pointer;
    }

    span {
        font-size: 12px;
        opacity: 0.5;
        text-transform: uppercase;
    }

    b {
        font-size: 13px;
    }

    h5 {
        font-weight: 400;
        font-size: 14px;
    }
}
