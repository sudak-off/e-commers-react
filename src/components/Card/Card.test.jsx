import { render, screen } from '@testing-library/react'
import Card from './Card'

describe('Card component', () => {
    it('Проверяем, рендерится ли компонент', () => {
        render(<Card />);
    });

    it('Есть ли в компоненте текст цена и грн', () => {
        render(<Card />);

        expect(screen.getByText('Цена:')).toBeInTheDocument();
        expect(screen.getByText(/ грн./)).toBeInTheDocument();

    });

    it('Есть тег img', () => {
        render(<Card />);

        expect(screen.getByRole('img')).toBeInTheDocument();
    });

    it('Не должно быть тега h1', () => {
        render(<Card />);

        expect(screen.queryByRole('list')).toBeNull();
    });

    it('Card snapshot', () => {
        const view = render(<Card price={2112} />);

        expect(view).toMatchSnapshot();
    });

    it('Card snapshot2', () => {
        const utils = render(<Card price={2122112} />);

        expect(utils).toMatchSnapshot();
    });
})