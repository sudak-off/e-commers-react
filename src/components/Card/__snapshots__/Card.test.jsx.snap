// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <img
          alt="shoes"
          height="135"
          width="100%"
        />
        <h5 />
        <div
          class="d-flex justify-between align-center"
        >
          <div
            class="d-flex flex-column"
          >
            <span>
              Цена:
            </span>
            <b>
              2 112
               грн.
            </b>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <img
        alt="shoes"
        height="135"
        width="100%"
      />
      <h5 />
      <div
        class="d-flex justify-between align-center"
      >
        <div
          class="d-flex flex-column"
        >
          <span>
            Цена:
          </span>
          <b>
            2 112
             грн.
          </b>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card component Card snapshot2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <img
          alt="shoes"
          height="135"
          width="100%"
        />
        <h5 />
        <div
          class="d-flex justify-between align-center"
        >
          <div
            class="d-flex flex-column"
          >
            <span>
              Цена:
            </span>
            <b>
              2 122 112
               грн.
            </b>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <img
        alt="shoes"
        height="135"
        width="100%"
      />
      <h5 />
      <div
        class="d-flex justify-between align-center"
      >
        <div
          class="d-flex flex-column"
        >
          <span>
            Цена:
          </span>
          <b>
            2 122 112
             грн.
          </b>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
